From 7de33fe46849946cf803e19d831d4a42390506cd Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Mon, 17 Aug 2020 17:09:14 +0000
Subject: [PATCH] Fix build with glibc-2.32

* sys_siglist was moved to compat symbol in:
  https://sourceware.org/git/?p=glibc.git;a=commit;h=b1ccfc061feee9ce616444ded8e1cd5acf9fa97f

* use implementation similar to:
  https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=66de2d4026054822aab89ac443e9bc726c9aae25

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 src/libSegFault/libSegFault.c.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/libSegFault/libSegFault.c.in b/src/libSegFault/libSegFault.c.in
index 883eca9..eb42bae 100755
--- a/src/libSegFault/libSegFault.c.in
+++ b/src/libSegFault/libSegFault.c.in
@@ -222,14 +222,14 @@ write_padded_number(int fd, unsigned long number, unsigned int base,
 static void
 write_strsignal(int fd, int signal)
 {
-	if (signal < 0 || signal >= _NSIG || _sys_siglist[signal] == NULL)
+	if (signal < 0 || signal >= _NSIG || strsignal(signal) == NULL)
 	{
 		WRITE_LITERAL("unknown signal: ");
 		write_number(fd, signal, 10);
 	}
 	else
 	{
-		WRITE_STRING(_sys_siglist[signal]);
+		WRITE_STRING(strsignal(signal));
 	}
 }
 
